app/views/site/render_global_gene_expression_plots.js.erb

```diff
@@ -78,22 +78,13 @@ $("#<%= @target %>-annotation, #<%= @target %>-subsample").change(function() {
 $("#<%= @target %>-cluster").change(function() {
     $('#cluster-plot').data('rendered', false);
     var newCluster = $(this).val();
-    var currAnnot = $('#<%= @target %>-annotation').val();
     // get new annotation options and re-render
     $.ajax({
         url: "<%= get_new_annotations_path(study_name: @study.url_safe_name)%>?cluster=" + encodeURIComponent(newCluster) + '&target=<%= @target %>',
         method: 'GET',
         dataType: 'script',
-        success: function (data) {
-            // parse response as a string and see if currently selected annotation exists in new annotations
-            if (data.indexOf(currAnnot) >= 0) {
-                $('#<%= @target %>-annotation').val(currAnnot);
-            }
-            $(document).ready(function () {
-                // since we now have new annotations, we need to set them in the search form for persistence
-                var an = $('#annotation').val();
-                renderDistribution<%= @identifier %>()
-            });
+        complete: function (jqXHR, textStatus) {
+            renderWithNewCluster(textStatus, renderDistribution<%= @identifier %>(), false);
```

Comment:
This `renderDistribution<%= @identifier %>()` is a holdover so I don't consider it blocking, but it'd be nice to refactor in the future to not construct the function name from a Rails variable.

Accepting `identifier` as a function parameter, e.g.:

```
renderDistribution(identifier);
```

Would enable disentangling JS and Rails.

Reply:
I think the reason we did that initially is because there is a dynamic function generated in the response to every call to load a gene result for global gene search, and each function needed to be specific to that response as there was a great deal of Ruby interpolation in setting the URLs, the cluster & annotation dropdowns, and other things - it wasn't just the value of `identifier`.  Those concerns aside, it would be possible to refactor into a generic `renderDistribution()` function with a little planning.  I think a refactoring is probably warranted since we will be likely mucking around in this code sooner rather than later as a part of advanced/faceted search work.